engines/kingdom/GPLogic1.cpp

```diff
@@ -0,0 +1,1501 @@
+		break;
+	}
+}
+
+void Logic::GPL1_32() {
+	_vm->processMapInput(3);
+}
+
+void Logic::GPL1_40() {
+	_nodeNum = 4;
+	_vm->drawLocation();
+	_vm->_userInput = 0;
+	if (_nodes[4] == 0)
+		_vm->playMovie(62);
+	if (_nodes[4] == 9) {
+		_vm->playMovie(8);
+		_vm->_tsIconOnly = true;
+	}
+	_currMap = 10;
+	_vm->playSound(14);
+	_statPlay = 41;
+}
+
+void Logic::GPL1_41() {
+	switch(_vm->_userInput) {
+	case 0x429:
+		_vm->playMovie(178);
+		_vm->playMovie(63);
+		inventoryDel(1);
+		_currMap = 10;
+		_nodes[4] = 9;
+		_statPlay = 41;
+		_vm->_tsIconOnly = true;
+		break;
+	case 0x43A:
+		_statPlay = 42;
+		_vm->_zoom = 2;
+		_vm->processMap(4, _vm->_zoom);
+		_vm->_userInput = 0;
+		break;
+	case 0x43E:
+		_vm->_sound = _vm->_lastSound;
+		_vm->playMovie(_vm->_pMovie);
+		break;
+	default:
+		if (_vm->_userInput)
+			debug("Skipped UserInput %d(0x%04X) for _StatPlay %d", _vm->_userInput, _vm->_userInput, _statPlay);
+		break;
+	}
+}
+
+void Logic::GPL1_42() {
+	_vm->processMapInput(4);
+}
+
+void Logic::GPL1_50() {
+	_nodeNum = 5;
+	_vm->drawLocation();
+	if (_nodes[5] != 0 && !_resurrect) {
+		_vm->_frameStop = 38;
+		_vm->playMovie(5);
+	}
+
+	if (_nodes[5] == 0)
+		_vm->playMovie(5);
+
+	_resurrect = false;
+	_vm->playSound(1);
+
+	if (_spell1 && _spell2 && _spell3) {
+		_vm->_sound = true;
+		_vm->playMovie(2);
+		_currMap = 10;
+		_statPlay = 51;
+		_vm->_loopFlag = true;
+	} else if (_vm->_daelonCntr) {
+		_vm->playMovie(2);
+		_currMap = 10;
+		_statPlay = 51;
+	} else {
+		switch(_nodes[5]) {
+			case 0:
+				inventoryAdd(1);
+				break;
+			case 1:
+				_vm->playMovie(60);
+				inventoryAdd(0);
+				_spell1 = true;
+				break;
+			case 2:
+				_vm->playMovie(59);
+				inventoryAdd(2);
+				_spell2 = true;
+				break;
+			case 3:
+				_vm->_sound = true;
+				_vm->playMovie(59);
+				inventoryAdd(3);
+				_spell3 = true;
+				break;
+			default:
+			    break;
+		}
+		if (_spell1 && _spell2 && _spell3) {
+			_vm->playMovie(6);
+			_currMap = 7;
+			_statPlay = 51;
+			_vm->_loopFlag = true;
+		} else {
+			_eye = false;
+			_vm->playMovie(61);
+			_vm->playSound(1);
+
+			if (!_spell1)
+				_vm->drawPic(116);
+
+			if (!_spell2)
+				_vm->drawPic(117);
+
+			if (!_spell3)
+				_vm->drawPic(118);
+
+			_currMap = 6;
+			_statPlay = 53;
+			_vm->_loopFlag = true;
+		}
+	}
+}
+
+void Logic::GPL1_51() {
+	switch(_vm->_userInput) {
+	case 0x406:
+		_statPlay = 70;
+		_vm->_loopFlag = true;
+		break;
+	case 0x40A:
+		_statPlay = 110;
+		_vm->_loopFlag = true;
+		break;
+	case 0x417:
+		_statPlay = 30;
+		_vm->_loopFlag = true;
+		break;
+	case 0x43A:
+		_statPlay = 52;
+		_vm->_zoom = 2;
+		_vm->processMap(5, _vm->_zoom);
+		_vm->_userInput = 0;
+		break;
+	case 0x43E:
+		_vm->_sound = _vm->_lastSound;
+		_vm->playMovie(_vm->_pMovie);
+		break;
+	default:
+		if (_vm->_userInput)
+			debug("Skipped UserInput %d(0x%04X) for _StatPlay %d", _vm->_userInput, _vm->_userInput, _statPlay);
```

Comment:
For all these warning messages you might want to consider using warning rather than debug. That'll give more clarity that the message is a warning

Reply:
done